src/Sentry.Unity/SystemInfoAdapter.cs

```diff
@@ -56,7 +57,32 @@ private SentrySystemInfoAdapter()
         public bool? SupportsVibration => SystemInfo.supportsVibration;
         public Lazy<string>? DeviceType => new(() => SystemInfo.deviceType.ToString());
         public string? CpuDescription => SystemInfo.processorType;
-        public string? DeviceName => SystemInfo.deviceName;
+        public string? DeviceName
+        {
+            get
+            {
+                // Workaround for https://github.com/getsentry/sentry-unity/issues/1322 -
+                if (Application.platform == RuntimePlatform.Android)
+                {
+                    using var version = new AndroidJavaClass("android.os.Build$VERSION");
+                    var sdkVersion = version.GetStatic<int>("SDK_INT");
+                    if (sdkVersion >= 32)
+                    {
+                        try
+                        {
+                            return SystemInfo.deviceName;
+                        }
+                        catch
+                        {
+                            // Unity's built-in fallback
+                            return "<unknown>";
+                        }
+                    }
+                }
+
+                return SystemInfo.deviceName;
```

Comment:
Maybe try-catch instead of checking a specific version? Also, it may just start working again with a higher SDK version, can't it? In that case this would also prevent returning the actual value.

Reply:
lol I'm blind. I've ignored the try-catch :D